### PR TITLE
Strengthen module types

### PR DESCRIPTION
**Pull request by jonludlam**
I had thought I'd already done this!

### CODE REVIEW

**Comment by jonludlam** [comment]
I think we might still be missing strengthening of class types too

File: src/xref2/strengthen.ml [hunk 1]
```diff
@@ -83,11 +83,7 @@ and module_ :
 and module_type :
     Cpath.module_type -> Component.ModuleType.t -> Component.ModuleType.t =
  fun prefix m ->
-  let expr =
-    match m.expr with
-    | None -> Some (ModuleType.Path { p_path = prefix; p_expansion = None })
-    | Some _ -> m.expr
-  in
+  let expr = Some (ModuleType.Path { p_path = prefix; p_expansion = None }) in
```
Comment (Julow) [inline src/xref2/strengthen.ml:86]: Should we keep `m.expr` into `p_expansion` ? What if `m.expr` is something complicated like `With` or `TypeOf` ?

**Review by Julow — APPROVED** [review]
LGTM

File: src/xref2/strengthen.ml (hunk 1 above)
Comment (jonludlam) [reply src/xref2/strengthen.ml:86]: Even if it _was_ something complicated now, it's simple now, and calculating the signature or expansion just looks at p_path and gets _it's_ expansion - [here](https://github.com/ocaml/odoc/blob/master/src/xref2/tools.ml#L1229-L1234) and [here](https://github.com/ocaml/odoc/blob/master/src/xref2/expand_tools.ml#L127-L131)

Comment (Julow) [reply src/xref2/strengthen.ml:86]: Alright, we don't need to keep the expansion around because it's getting recomputed anyway.

**Pull request merged**